### PR TITLE
Allow trailing slash in permutePath

### PR DESCRIPTION
**Pull request by ruoho**

### CODE REVIEW

File: test/domain_and_path_test.js [hunk 1]
```diff
@@ -187,7 +187,7 @@ vows
       "trailing slash": {
         topic: tough.permutePath.bind(null, '/foo/bar/'),
         "three things": function (list) {
```
Comment (awaterma) [inline test/domain_and_path_test.js]: Change the test title to "four things"?

Comment (ruoho) [reply test/domain_and_path_test.js]: Updated

File: test/domain_and_path_test.js [hunk 2]
```diff
@@ -186,8 +186,8 @@ vows
       },
       "trailing slash": {
         topic: tough.permutePath.bind(null, '/foo/bar/'),
-        "three things": function (list) {
-          assert.deepEqual(list, ['/foo/bar', '/foo', '/']);
+        "four things": function (list) {
+          assert.deepEqual(list, ['/foo/bar/', '/foo/bar', '/foo', '/']);
```
Comment (stash) [inline test/domain_and_path_test.js]: Perhaps `/foo/` should be an intermediate as well? 

Overall, I think I have a performance concern: is this is going to result in a lot more hits to the underlying store?

**Comment by stash** [comment]
Is this resolving #111 ?

**Comment by awaterma** [comment]
We need to ensure this is correct, according to the RFC. I think that should be the determinant here. If the RFC is not clear, we will support the current behavior.

By RFC, we should support the BIS-Draft, version 3.

**Pull request closed without merging**